tasks/snyk-code-test.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: snyk-code-test
+spec:
+  description: >-
+    Static code security test with snyk 
+  workspaces:
+    - name: workspace
+      optional: true
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  params:
+    - name: shared-secret
+      default: test-team-snyk
+    - name: args
+      type: string
+      description: extra args needs to append
+      default: "--all-projects --exclude=test*,vendor,deps"
+  volumes:
+    - name: snyk-secret
+      csi:
+        readOnly: true
+        driver: csi.sharedresource.openshift.io
+        volumeAttributes:
+          sharedSecret: $(params.shared-secret)
+  steps:
+    - name: snyk-code-test 
+      image: quay.io/jinqi/hacbstest_jinqi_1:latest
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+        - name: snyk-secret
+          mountPath: "/etc/secrets"
+          readOnly:true
+      script: |
+        #!/usr/bin/env bash
+        SNYK_TOKEN="$(cat /etc/secrets/snyk_token)" 
```

Comment:
nit: handle situation directly when SNYK_TOKEN is empty.

Reply:
It can be handled by snyk cmd and  it will be reported as an error in the end of the task.

Reply:
it depends what it will report if it will be "No password provided" then it's fine, but if it will return "Unauthorized" then it's not clear if the secret is provided properly.

Reply:
The  error message is "`snyk` requires an authenticated account. Please run `snyk auth` and try again." and updated it to prompt a message for creating a secret 'test-team-snyk' .